rasa_core/tracker_store.py

```diff
@@ -250,3 +256,137 @@ def retrieve(self, sender_id):
 
     def keys(self):
         return [c["sender_id"] for c in self.conversations.find()]
+
+
+class SQLTrackerStore(TrackerStore):
+    """Store which can save and retrieve trackers from an SQL database"""
+
+    from sqlalchemy.ext.declarative import declarative_base
+    Base = declarative_base()
+
+    class SQLEvent(Base):
+        __tablename__ = 'conversations'
+
+        id = Column(Integer, primary_key=True)
+        sender_id = Column(String, nullable=False)
+        type_name = Column(String, nullable=False)
+        timestamp = Column(Float)
+        intent_name = Column(String)
+        action_name = Column(String)
+        data = Column(String)
+
+    def __init__(self,
+                 domain: Optional[Domain] = None,
+                 dialect: Text = 'sqlite',
+                 host: Text = None,
+                 event_broker: Optional[EventChannel] = None,
+                 db: Text = 'rasa.db',
+                 username: Text = None,
+                 password: Text = None) -> None:
+        from sqlalchemy.orm import sessionmaker
+        from sqlalchemy.engine.url import URL
+        from sqlalchemy import create_engine
+
+        engine_url = URL(dialect, username, password, host, database=db)
+
+        logger.debug('Attempting to connect to database '
+                     'via "{}"'.format(engine_url.__to_string__()))
+
+        self.engine = create_engine(engine_url)
+        self.session = sessionmaker(bind=self.engine)()
+
+        logger.debug('Connection successful, ensuring conversations table...')
```

Comment:
```suggestion
        logger.debug('Connection to SQL db successful, ensuring conversations table exists...')
```

Reply:
this logging stmt should be moved into the `_ensure_table` function

Reply:
OK, but I'm not sure about putting "exists" - if the table doesn't exist, then `_ensure_table` will create it